src/main/kotlin/no/skatteetaten/aurora/gobo/integration/cantus/ImageRegistryServiceBlocking.kt

```diff
@@ -3,65 +3,100 @@ package no.skatteetaten.aurora.gobo.integration.cantus
 import mu.KotlinLogging
 import no.skatteetaten.aurora.gobo.ServiceTypes
 import no.skatteetaten.aurora.gobo.TargetService
-import no.skatteetaten.aurora.gobo.resolvers.blockNonNullAndHandleError
+import no.skatteetaten.aurora.gobo.integration.SourceSystemException
+import no.skatteetaten.aurora.gobo.resolvers.handleError
+import no.skatteetaten.aurora.gobo.resolvers.imagerepository.ImageTag
 
 import org.springframework.stereotype.Service
+import org.springframework.util.LinkedMultiValueMap
 import org.springframework.web.reactive.function.client.WebClient
 import org.springframework.web.reactive.function.client.bodyToMono
+import org.springframework.web.util.UriComponentsBuilder
+import reactor.core.publisher.Mono
+import reactor.core.publisher.toMono
 
 private val logger = KotlinLogging.logger {}
 
+data class ImageRepoAndTags(val imageRepository: String, val imageTags: List<String>) {
+    fun getTagUrls() = imageTags.map { "$imageRepository/$it" }
+
+    companion object {
+        fun fromImageTags(imageTags: Set<ImageTag>) =
+            imageTags.groupBy { it.imageRepository.repository }.map { entry ->
+                val imageTagStrings = entry.value.map { it.name }
+                ImageRepoAndTags(entry.key, imageTagStrings)
+            }
+    }
+}
+
+private fun List<ImageRepoAndTags>.getAllTagUrls() =
+    this.flatMap { it.getTagUrls() }
+
 @Service
 class ImageRegistryServiceBlocking(
     @TargetService(ServiceTypes.CANTUS) val webClient: WebClient
 ) {
 
-    fun resolveTagToSha(imageRepoDto: ImageRepoDto, imageTag: String, token: String) =
-        getAuroraResponseImageTagResource(imageRepoDto, imageTag, token).dockerDigest
+    fun resolveTagToSha(imageRepoDto: ImageRepoDto, imageTag: String, token: String): String? {
+        val auroraImageTagResource: AuroraResponse<ImageTagResource> =
+            execute<AuroraResponse<ImageTagResource>>(token) {
+                logger.debug("Retrieving type=ImageTagResource from  url=${imageRepoDto.registry} image=${imageRepoDto.imageName}/$imageTag")
+                it.get().uri(
+                    "/manifest?tagUrls=${imageRepoDto.registry}/{namespace}/{imageTag}/{tag}",
+                    imageRepoDto.mappedTemplateVars.plus("tag" to imageTag)
+                )
+            }.block()!!
+        return ImageTagDto.toDto(auroraImageTagResource, imageTag, imageRepoDto).dockerDigest
+    }
 
-    fun findTagByName(
-        imageRepoDto: ImageRepoDto,
-        imageTag: String,
+    fun findTagsByName(
+        imageReposAndTags: List<ImageRepoAndTags>,
         token: String
-    ) = getAuroraResponseImageTagResource(imageRepoDto, imageTag, token)
+    ): AuroraResponse<ImageTagResource> {
+        val tagUrlsQueryParameters =
+            LinkedMultiValueMap<String, String>().apply { addAll("tagUrls", imageReposAndTags.getAllTagUrls()) }
+
+        val tagPath = UriComponentsBuilder
+            .fromPath("/manifest")
+            .queryParams(tagUrlsQueryParameters)
```

Comment:
Nå så bruker ikke query parametre placeholders for URI. Jeg kan ikke se noen løsning på dette akkurat nå. 
La oss si at dette er query parametrene:
tagUrls=docker.com/{namespace}/{name}/{tag}&tagUrls=docker.com/{namespace}/{name}/{tag}

Dette filer når jeg prøver å legge til urivariabler. Noen som har innspill eller forslag? 
Bruker UriComponentsBuilder og uriVariables.

Reply:
Hva er feilen du får?

Reply:
Det er helt greit. Micrometer ignorerer query parameters uansett.

Reply:
Jeg prøvde å bruke UriComponentsBuilder.UriVariables(Map<String, List<String>> men da replacet den placeholderen name med hele listen. Prøvde også UriComponentsBuilder.Build(List<String>),men da klarte den ikke å finne en verdier for placeholderne. 

Dette er ikke et veldig stort problem. Det beste hadde jo vært å få det til slik at man får gode metrikker.

Reply:
Denne er løst. Micrometer ignorerer queryparametre.